### PR TITLE
Improve development instructions

### PR DESCRIPTION
**Pull request by FedericoGarza**
### Pull Request Description

I've noticed that the minimum Python version required is `3.8`. I've added this information in this PR and instructions to set up a Python environment using `conda`. 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ We currently utilize GitHub Issues as our project management tool for EvalML. Pl
 
 
 #### 1. Clone repo
-The code is hosted on GitHub, so you will need to use Git to clone the project and make changes to the codebase. Once you have obtained a copy of the code, you should create a development environment that is separate from your existing Python environment so that you can make and test changes without compromising your own work environment.
+The code is hosted on GitHub, so you will need to use Git to clone the project and make changes to the codebase. Once you have obtained a copy of the code, you should create a development environment that is separate from your existing Python environment so that you can make and test changes without compromising your own work environment. Additionally, you must make sure that the version of Python you use is at least 3.8. Using `conda` you can use `conda create -n evalml python=3.8` and `conda activate evalml` before the following steps.
```
Comment (freddyaboulton) [inline contributing.md:14]: Thank you @FedericoGarza !

**Comment by chukarsten** [comment]
Thanks for contributing @FedericoGarza !  Please take a minute to update the release notes with this PR number and that should pass your CI!  Let me know if you have any questions.  Be happy to help.

**Comment by FedericoGarza** [comment]
> Thanks for contributing @FedericoGarza ! Please take a minute to update the release notes with this PR number and that should pass your CI! Let me know if you have any questions. Be happy to help.

Hi @chukarsten. I've added the PR to the release notes in [57f9aa4](https://github.com/alteryx/evalml/pull/3468/commits/57f9aa45111be66f14815f44a1b4c05bcc0b9c55). Let me know if anything else is needed. :)

**Review by chukarsten — APPROVED** [review]
Thanks a ton for contributing!

**Pull request merged**